modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/GeometryColumnEncoder.java

```diff
@@ -24,19 +24,22 @@
 
     private final boolean atLeast2_2_0;
     private final boolean stSimplifyEnabled;
+    private final boolean stPreserveTopologyEnabled;
     private final boolean encodeBase64;
     private final PostGISDialect dialect;
 
     GeometryColumnEncoder(
             Version version,
             boolean stSimplifyEnabled,
+            boolean stPreserveTopologyEnabled,
```

Comment:
If we are going to keep this up we should break out an enum with different options for how to simplify.

Reply:
I thought the same. However, up to what level should it be brought? Here? Dialect? Factory?
If factory, then backwards compatibility for old settings should be accounted for (if the simplification mode is missing, look for the legacy "simplify" flag and use that).

Reply:
@aaime @jodygarnett maybe we can (for backward compatibility) leave the simplify flag there and just add the enum to choose between ST_Simplify and ST_SimplifyPreserveTopology if simplify is true.

Reply:
@mbarto works for me

Reply:
@aaime I added the SimplificationMethod enum to replace the boolean flag with options FAST (ST_Simplify) and PRESERVETOPOLOGY